matching-notification-resources/src/main/resources/resources/uicomponents/matchingNotification/matchesTable.js

```diff
@@ -1007,21 +1065,123 @@ var PhenoTips = (function (PhenoTips) {
-        if (this._isAdmin && serverId == '' && emails.length > 0) {
-            td += '<span class="fa fa-envelope" title="' + this._NOTIFY + '"></span> <input type="checkbox" class="notify" data-matchid="' + matchId + '" data-patientid="'+ patientId +'" data-emails="'+ emails.toString() +'">';
+        if (this._isAdmin && patient.serverId == '' && patient.emails.length > 0) {
+            td += '<span class="fa fa-envelope" title="' + this._NOTIFY + '"></span> <input type="checkbox" class="notify" data-matchid="' + matchId + '" data-patientid="'+ patient.patientId +'" data-emails="'+ patient.emails.toString() +'">';
         }
         td += '</td>';
         return td;
     },
 
-    _getContact : function(record)
+    _generateNotificationHistoryTable : function(records)
+    {
+        var table = '<table class="notification-history-table"><tbody>';
+
+        var header = '<tr>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_DATE + '</th>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_FROM + '</th>';
+        header += '<th scope="col">' + this._NOTIFICATION_HISTORY_TO + '</th>';
+        header += '</tr>';
+
+        table += header;
+
+        for (var i=0; i < records.length; i++) {
+            var record = records[i];
+            var row = '<tr class="' + record.type + '" title="' + record.type + '">';
+
+            var date = '<td>';
+            if (record.date) {
+                var dateIconName = record.type == 'contact' ? 'fa fa-envelope-o' : 'fa fa-volume-up' ;
+                date += '<div class="date">' + record.date.split(' ')[0] + '</div><span class="'+ dateIconName + '"> </span></td>';
+            }
+            date += '</td>';
+
+            var from = '<td>';
+            if (record.from) {
+                from += this._generateRecipientCell(record.from, record.type);
+            }
+            from += '</td>';
+
+            var to = '<td>';
+            if (record.to) {
+                to += this._generateRecipientCell(record.to, record.type);
+            }
+            to += '</td>';
+
+            row += date;
+            row += from;
+            row += to;
+            row += '</tr>';
+
+            table += row;
+        }
+
+        table += '</tbody></table>';
+
+        return table;
+    },
+
+    _generateRecipientCell : function(info, type)
+    {
+        // notifications should have something generic e.g. "PhenomeCentral" (no link)
+        if (type == "notification" && info.userinfo.id == "xwiki:XWiki.Admin") {
+            return "PhenomeCentral";
+        }
+
+        var cell = '';
+        // user name and institution
+        if (info.userinfo) {
+            if (info.userinfo.name) {
+                if (info.userinfo.id) {
+                    var space = (info.userinfo.id.startsWith("xwiki:XWiki.")) ? "XWiki" : "Groups";
```

Comment:
This is brittle. Please either don't link to tye user's profile or send the url in the JSON instead of generating it here.

Reply:
This will not work for Brittney nor any other admin 
![image](https://user-images.githubusercontent.com/23243244/51777901-315be000-20cd-11e9-8d3a-6e923930c639.png)

I believe it is causing problem shown below  where Peter is an administrator sending notification (I will log it separatelly)
![image](https://user-images.githubusercontent.com/23243244/51777956-76801200-20cd-11e9-9c36-67a1755dc45a.png)